### PR TITLE
Add playbook entry for CortexGossipMembersMismatch.

### PR DESCRIPTION
**Pull request by stevesg**
**What this PR does**:
Add playbook entry for CortexGossipMembersMismatch.

**Checklist**
- [x] ~~`CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`~~

### CODE REVIEW

File: cortex-mixin/docs/playbooks.md [hunk 1]
```diff
@@ -572,7 +572,31 @@ How to **fix**:
 
 ### CortexGossipMembersMismatch
 
-_TODO: this playbook has not been written yet._
+This alert fires when any instance does not register all other instances as members of the memberlist cluster.
+
+How it **works**:
+- This alert applies when memberlist is used for the ring backing store.
+- All Cortex instances, regardless of type, join the a single memberlist cluster.
```
Comment (pracucci) [inline cortex-mixin/docs/playbooks.md:579]:
```suggestion
- All Cortex instances, regardless of type, join a single memberlist cluster.
```

File: cortex-mixin/docs/playbooks.md [hunk 2]
```diff
@@ -572,7 +572,31 @@ How to **fix**:
...
+- Each instance (=memberlist cluster member) should be able to see all others.
+- Therefore the following should be equal for every instance:
+  - The reported number of cluster members (`memberlist_client_cluster_members_count`)
+  - The total number of currently responsive instances.
```
Comment (pracucci) [inline cortex-mixin/docs/playbooks.md:583]: I've noticed the alert doesn't include `query-frontend` and `query-scheduler` in the `up` count. Should it or they're not part of the memberlist cluster?

File: cortex-mixin/docs/playbooks.md [hunk 3]
```diff
@@ -572,7 +572,31 @@ How to **fix**:
...
+
+How to **investigate**:
+- The instance which has the incomplete view of the cluster (too few members) is specified in the alert.
+- If the count is zero:
+  - It is possible that the joining the cluster has yet to succeed.
+  - The following log message indicates that the _initial_ initial join did not succeed: `failed to join memberlist cluster`
+  - The following log messages indicate that subsequent re-join attempts are failing: `re-joining memberlist cluster failed`
```
Comment (pracucci) [inline cortex-mixin/docs/playbooks.md:590]: [nit] "The following log messages" but you mention 1 message (re-joining memberlist cluster failed). Maybe I don't understand what you mean here? 🤔

**Review by pracucci — APPROVED** [review]
Lovely playbook! Makes sense to me. I just left a couple of questions.

File: cortex-mixin/docs/playbooks.md (hunk 2 above)
Comment (stevesg) [reply cortex-mixin/docs/playbooks.md:583]: Not sure, I'll check. I think anything that starts the memberlist service will join (regardless if they use the ring or not).

File: cortex-mixin/docs/playbooks.md (hunk 3 above)
Comment (stevesg) [reply cortex-mixin/docs/playbooks.md:590]: Ah, typo! :grin:

File: cortex-mixin/docs/playbooks.md (hunk 2 above)
Comment (stevesg) [reply cortex-mixin/docs/playbooks.md:583]: Checked this: AFAICT they do not start the MemberlistKV service, so will not join the cluster.

Comment (pracucci) [reply cortex-mixin/docs/playbooks.md:583]: Perfect. Then alert is correct! 👍

**Pull request merged**